DelvUI/Interface/AstrologianHudWindow.cs

```diff
@@ -1,121 +1,81 @@
-        private Dictionary<string, uint> StarGiantColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 5];
-        private Dictionary<string, uint> LightspeedColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 6];
-        private Dictionary<string, uint> DotColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 7];
-        private Dictionary<string, uint> StarGlowColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 8];
-        private Dictionary<string, uint> DivinationGlowColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 9];
-        private Dictionary<string, uint> DrawMeleeGlowColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 10];
-        private Dictionary<string, uint> DrawRangedGlowColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 11];
-        private Dictionary<string, uint> DrawCDColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 12];
-        private Dictionary<string, uint> DrawCDReadyColor => PluginConfiguration.JobColorMap[Jobs.AST * 1000 + 13];
-
-        protected override void Draw(bool _) {
-            if (ShowDivinationBar) {
+        public AstrologianHudWindow(DalamudPluginInterface pluginInterface, PluginConfiguration pluginConfiguration,
+            AstrologianHudConfig config) :
+            base(pluginInterface, pluginConfiguration)
+        {
+            _config = config;
+        }
+
+        protected override void Draw(bool _)
+        {
+            if (_config.ASTShowDivinationBar)
+            {
                 DrawDivinationBar();
             }
 
-            if (ShowDrawBar) {
+            if (_config.ASTShowDrawBar)
+            {
                 DrawDraw();
             }
 
-            if (ShowDotBar) {
+            if (_config.ASTShowDotBar)
+            {
                 DrawDot();
             }
 
-            if (ShowLightspeedBar) {
+            if (_config.ASTShowLightspeedBar)
+            {
                 DrawLightspeed();
             }
 
-            if (ShowStarBar) {
+            if (_config.ASTShowStarBar)
+            {
                 DrawStar();
             }
         }
 
-        protected override void DrawPrimaryResourceBar() {
-            if (!ShowPrimaryResourceBar) {
+        protected override void DrawPrimaryResourceBar()
+        {
+            if (!_config.ASTShowPrimaryResourceBar)
+            {
                 return;
             }
 
             base.DrawPrimaryResourceBar();
         }
 
-        private void DrawDivinationBar() {
-            var chunkColors = new List<Dictionary<string, uint>>();
-
-            if (PluginInterface.ClientState.LocalPlayer?.ClassJob.Id != 33) {
+        private void DrawDivinationBar()
+        {
+            List<Dictionary<string, uint>> chunkColors = new List<Dictionary<string, uint>>();
+            if ((PluginInterface.ClientState.LocalPlayer?.ClassJob.Id) != 33)
```

Comment:
Curious on why you need to check the job id here.
Supposedly this is being taken care of before this class gets told to draw.
Also if this is really necessary, please replace the hardcoded number by using the Jobs enum.

Reply:
It's was a safety check, since there is an unsafe code bit after that.
I removed it since it's being taken care of before the draw.